src/app/content/highlights/components/cardUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { Highlight } from '@openstax/highlighter';
+import { HTMLElement, } from '@openstax/types/lib.dom';
+import React from 'react';
+import { findElementSelfOrParent } from '../../../domUtils';
+import { assertWindow } from '../../../utils';
+
+export const useDebouncedWindowSize = () => {
+  const window = assertWindow();
+  const timeout = React.useRef(0);
+  const [size, setSize] = React.useState([0, 0]);
+
+  React.useLayoutEffect(() => {
+    const updateSize = () => {
+      clearTimeout(timeout.current);
+      timeout.current = setTimeout(() => {
+        setSize([window.innerWidth, window.innerHeight]);
+      }, 50);
+    };
+    window.addEventListener('resize', updateSize);
+    updateSize();
+    return () => window.removeEventListener('resize', updateSize);
```

Comment:
I think you should also clear the timeout in the clean up function.

Reply:
I think this cleanup function is called only when a component unmounts which mean that this state should be cleared anyway, right?

Reply:
oh, wait, will it still try to call `setSize` after unmounting?
This will probably never cause an error because it would be hard to unmount this component while resizing window, but still, it may happen, nice catch Jan ;)

Reply:
It will remove the listener, but the timeout might still be running -> attempt at updating the state of an unmounted component, (even though the state is theoretically in a hook) which gives the `warning: can't perform state update on an unmounted component'.